app/models/user.rb

```diff
@@ -5,6 +5,50 @@ class User < ApplicationRecord
   include Searchable
   include Storext.model
 
+  # NOTE: @citizen428 This is temporary code during profile migration and will
+  # be removed.
+  # rubocop:disable Metrics/BlockLength
+  concerning :ProfileMigration do
+    included do
+      PROFIE_FIELDS =
+        (Profiles::ExtractData::DIRECT_ATTRIBUTES + Profiles::ExtractData::MAPPED_ATTRIBUTES.values).freeze
+
+      # All new users should automatically have a profile
+      after_create_commit { Profile.create(user: self, data: Profiles::ExtractData.call(self)) }
+
+      # Keep saving changes locally for the time being, but propagate them to profiles.
+      after_update_commit do
+        if (previous_changes.keys.map(&:to_sym) & User::PROFIE_FIELDS).present?
```

Comment:
Only update the profile if the recent changes actually included a profile field.